docs/API.md

```diff
@@ -104,133 +177,185 @@ gulp.task('somename', function() {
 
 #### name
 
-The name of the task. Tasks that you want to run from the command line should not have spaces in them.
+Optional, The name of the task. Tasks that you want to run from the command line
+should not have spaces in them.
 
-#### deps
-Type: `Array`
+If the name is not provided, the task will be name after the function `name` attribute.
 
-An array of tasks to be executed and completed before your task will run.
+`Function.name` is not writable; it cannot be set or edited.
+It will be empty for anonymous function:
 
 ```js
-gulp.task('mytask', ['array', 'of', 'task', 'names'], function() {
-  // Do stuff
-});
+// foo.name === 'foo'
+function foo() {};
+
+// bar.name === ''
+var bar = function() {};
+
+// bar.name === ''
+bar.name = 'bar'
 ```
 
-**Note:** Are your tasks running before the dependencies are complete?  Make sure your dependency tasks are correctly using the async run hints: take in a callback or return a promise or event stream.
+You should either provide the task name or avoid anonymous functions.
 
 #### fn
 
-The function that performs the task's operations. Generally this takes the form of `gulp.src().pipe(someplugin())`.
+The function that performs the task's operations.
+Generally this takes the form of `return gulp.src().pipe(someplugin());`.
 
-#### Async task support
+Gulp assumes each task is asynchronous and call them with a callback parameter.
+A Task should either call the callback parameter to signal when operations
+have ended, or return a stream, promise, child process or RxJS observable.
 
-Tasks can be made asynchronous if its `fn` does one of the following:
+#### Async support
 
 ##### Accept a callback
 
-```javascript
-// run a command in a shell
-var exec = require('child_process').exec;
-gulp.task('jekyll', function(cb) {
-  // build Jekyll
-  exec('jekyll build', function(err) {
-    if (err) return cb(err); // return error
-    cb(); // finished task
-  });
+```js
+var del = require('del');
+
+gulp.task('clean', function(done) {
+  del(['.build/'], done);
 });
 ```
 
+The callback accept an optional `Error` object. If it receives an error,
```

Comment:
@phated the callback signature is `callback(error, result)` I think. Does the result parameter matter in Gulp?